persistence/trees/trees.go

```diff
@@ -0,0 +1,574 @@
+		if err != nil {
+			return err
+		}
+
+		actorBz, err := codec.GetCodec().Marshal(actor)
+		if err != nil {
+			return err
+		}
+
+		merkleTreeName, ok := actorTypeToMerkleTreeName[actorType]
+		if !ok {
+			return fmt.Errorf("no merkle tree found for actor type: %s", actorType)
+		}
+		if err := t.merkleTrees[merkleTreeName].Update(bzAddr, actorBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+//////////////////////////
+// Account Tree Helpers //
+//////////////////////////
+
+func (t *treeStore) updateAccountTrees(accounts []*coreTypes.Account) error {
+	for _, account := range accounts {
+		bzAddr, err := hex.DecodeString(account.GetAddress())
+		if err != nil {
+			return err
+		}
+
+		accBz, err := codec.GetCodec().Marshal(account)
+		if err != nil {
+			return err
+		}
+
+		if err := t.merkleTrees[accountMerkleTree].Update(bzAddr, accBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *treeStore) updatePoolTrees(pools []*coreTypes.Account) error {
+	for _, pool := range pools {
+		bzAddr, err := hex.DecodeString(pool.GetAddress())
+		if err != nil {
+			return err
+		}
+
+		accBz, err := codec.GetCodec().Marshal(pool)
+		if err != nil {
+			return err
+		}
+
+		if err := t.merkleTrees[poolMerkleTree].Update(bzAddr, accBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+///////////////////////
+// Data Tree Helpers //
+///////////////////////
+
+func (t *treeStore) updateTransactionsTree(indexedTxs []*coreTypes.IndexedTransaction) error {
+	for _, idxTx := range indexedTxs {
+		txBz := idxTx.GetTx()
+		txHash := crypto.SHA3Hash(txBz)
+		if err := t.merkleTrees[transactionsMerkleTree].Update(txHash, txBz); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (t *treeStore) updateParamsTree(params []*coreTypes.Param) error {
+	for _, param := range params {
+		paramBz, err := codec.GetCodec().Marshal(param)
+		paramKey := crypto.SHA3Hash([]byte(param.Name))
+		if err != nil {
+			return err
+		}
+		if err := t.merkleTrees[paramsMerkleTree].Update(paramKey, paramBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *treeStore) updateFlagsTree(flags []*coreTypes.Flag) error {
+	for _, flag := range flags {
+		flagBz, err := codec.GetCodec().Marshal(flag)
+		flagKey := crypto.SHA3Hash([]byte(flag.Name))
+		if err != nil {
+			return err
+		}
+		if err := t.merkleTrees[flagsMerkleTree].Update(flagKey, flagBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getActorsUpdated is responsible for fetching the actors that have been updated at a given height.
+func (t *treeStore) getActorsUpdated(
+	pgtx pgx.Tx,
+	actorType coreTypes.ActorType,
+	height uint64,
+) ([]*coreTypes.Actor, error) {
+	actorSchema, ok := actorTypeToSchemaName[actorType]
+	if !ok {
+		return nil, fmt.Errorf("no schema found for actor type: %s", actorType)
+	}
+
+	// TECHDEBT (ISSUE #813): Avoid this cast to int64
+	// https://github.com/pokt-network/pocket/issues/813
```

Comment:
Leaving one comment but ditto elsewhere 

```suggestion
	// TECHDET(#813): Avoid this cast to int64
```